lib/dynflow/action/polling.rb

```diff
@@ -13,6 +14,9 @@ def run(event = nil)
         end
       when Poll
         poll_external_task_with_rescue
+      when Timeout
+        process_timeout
+        poll_external_task_with_rescue
```

Comment:
shouldn't that then be part of the overridden process_timeout the responsibility to poll?